[REEF-620]  Adding language property in Avro Configuration

jwang98052 opened the pull request:
When the code in a language receives a configuration from another, it needs to know which language the configuration was created so that to determine which alias to choose.

This PR added a language property in Avro Configuration to carry this information. It also modified the places where the AvroConfiguration object is created in both Java and C#.

The evaluator.conf used in testing is a newly generated one from test runtime which contains language information. Test is updated to read language property from it after deserialization.

JIRA: REEF-620(https://issues.apache.org/jira/browse/REEF-620)

This closes #

Author: Julia Wang  Email: juliaw@apache.org

markusweimer commented on lang/java/reef-tang/tang/src/main/avro/configuration.avsc:
```diff
@@ -31,6 +31,7 @@
     "type":"record",
     "name":"AvroConfiguration",
     "fields":[
+    {"name":"language","type":"string"},
```
Nitpick: The indent is off compared to the rest of the fields.

jwang98052 replied:
The original one doesn't have indent. But I can make indent for all of them.

markusweimer commented on lang/cs/Org.Apache.REEF.Evaluator.Tests/EvaluatorConfigurationsTests.cs:
```diff
@@ -46,9 +44,9 @@ public void TestEvaluatorConfigurations()
             Logger.Log(Level.Info, "ErrorHandlerRID = " + rId);
 
-            Assert.IsTrue(eId.Equals("Node-1-1437686223482"));
+            Assert.IsTrue(eId.Equals("Node-1-1440108430564"));
```
These changes seem unrelated to the goals of the PR. Can you please explain?

jwang98052 replied:
Since I changed AvroConfiguration, I got new generated evaluator.conf for testing which has different data.

markusweimer replied:
Ah, I see

markusweimer commented on lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestAvroConfiguration.cs:
```diff
@@ -62,7 +62,7 @@ private AvroConfiguration ToAvroConfiguration()
             b.Add(e1);
             b.Add(e2);
-            var a = new AvroConfiguration(b);
+            var a = new AvroConfiguration("CS", b);
```
Would it make sense to have an `enum` defined in Avro for the languages? That would be safer than magic strings, but I don't know how feasible it is.

jwang98052 replied:
Make sense. Will do it.

markusweimer commented:
@jwang98052 I've done a pass. I have some broader concerns I'll put onto the JIRA.

jwang98052 commented:
Addressed review comments, merged from master and rebased it as more check-in happened yesterday. 
For Avro schema, I tried to use space to replace tab, but git doesn't find the difference. 
Change pushed in.

markusweimer commented:
Hmm, the indent still looks off, even [here on GitHub](https://github.com/Microsoft-CISL/incubator-reef/blob/REEF-620/lang/java/reef-tang/tang/src/main/avro/configuration.avsc)

jwang98052 commented:
Would you like to help me to update it before merging?

markusweimer commented:
Yes, will do. Otherwise, this looks good. Will merge it.
